INSS.EIIR.Functions/Functions/ExtractJobNotificationTrigger.cs

```diff
@@ -0,0 +1,87 @@
+using INSS.EIIR.Interfaces.DataAccess;
+using INSS.EIIR.Interfaces.Messaging;
+using INSS.EIIR.Models.Configuration;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace INSS.EIIR.Functions.Functions
+{
+    public class ExtractJobNotificationTrigger
+    {
+        private readonly ILogger<ExtractJobNotificationTrigger> _logger;
+        private readonly IServiceBusMessageSender _serviceBusMessageSender;
+        private readonly IExtractRepository _eiirRepository;
+        private readonly ServiceBusConfig _serviceBusConfig;
+        private readonly NotifyConfig _notifyConfig;
+
+        public ExtractJobNotificationTrigger(
+            ILogger<ExtractJobNotificationTrigger> log,
+            IOptions<ServiceBusConfig> sbOptions,
+            IOptions<NotifyConfig> nfOptions,
+            IServiceBusMessageSender serviceBusMessageSender,
+            IExtractRepository eiirRepository)
+        {
+            _logger = log;
+            _serviceBusConfig = sbOptions.Value;
+            _notifyConfig = nfOptions.Value;
+            _serviceBusMessageSender = serviceBusMessageSender;
+            _eiirRepository = eiirRepository;
+        }
+
+        [FunctionName("ExtractJobNotificationTrigger")]
+        public async Task Run([BlobTrigger("%blob:containername%/{name}.zip", Connection = "blob:connectionstring")] byte[] myBlob, string name, Uri uri)
+        {
+            string message = $"ExtractJobNotificationTrigger Blob trigger function triggered for blob\n Name: {name}  \n with uri:{uri.AbsoluteUri}";
+            _logger.LogInformation(message);
+
+            if (string.IsNullOrEmpty(_serviceBusConfig.NotifyQueue))
+            {
+                _logger.LogError("ExtractjobTrigger missing configuration servicebus:notifyqueue");
+                throw new Exception("ExtractjobTrigger missing configuration servicebus:notifyqueue");
+            }
+
+            if (string.IsNullOrEmpty(_notifyConfig.SubscriberEmailTemplateId))
+            {
+                _logger.LogError("ExtractjobTrigger missing configuration subscriberEmailTemplateId");
```

Comment:
duplicate strings here and elsewhere. Can these go as either vars or string templates at the class level

Reply:
Agreed will move, I had planned too but just rushing as usual.